File: backend/Models/Note.cs
namespace backend.Models
{
    public class Note
    {
        public int Id { get; set; }
        public string UserName { get; set; } = string.Empty;
        public string Title { get; set; } = string.Empty;
        public string? Content { get; set; }
        public DateTime CreatedAt { get; set; }
        public DateTime UpdatedAt { get; set; }
    }

    // What the client sends to CREATE a note
    public class CreateNoteDto
    {
        public string Title { get; set; } = string.Empty;
        public string? Content { get; set; }
    }

    // What the client sends to UPDATE a note
    public class UpdateNoteDto
    {
        public string Title { get; set; } = string.Empty;
        public string? Content { get; set; }
    }
}

File: frontend/src/types/note.ts
export interface Note {
  id: number
  title: string
  content: string | null
  createdAt: string
  updatedAt: string
}

export interface CreateNoteDto {
  title: string
  content: string | null
}

export interface UpdateNoteDto {
  title: string
  content: string | null
}